mobile/library/common/internal_engine.cc

```diff
@@ -174,6 +177,7 @@ envoy_status_t InternalEngine::main(std::shared_ptr<Envoy::OptionsImplBase> opti
                                                         server_->api().randomGenerator());
           dispatcher_->drain(server_->dispatcher());
           callbacks_->on_engine_running_();
+          engine_running_.Notify();
```

Comment:
should this be added before the on_engine_running callback, in case e.g. that on_engine_running callback happens to block?

Reply:
Good point. Updated.